### PR TITLE
fix(amazonq): better location for inline chat editor hint

### PR DESCRIPTION
**Pull request by zixlin7**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description
make the popup show at line end with some offset so it doesn't block code
- before:
Screenshot 2024-11-06 at 3 46 57 PM

- after:
Screenshot 2024-11-06 at 4 05 55 PM

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by Will-ShaoHua** [comment]
@zixlin7 could you put a screenshot of updated ui

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/inline/InlineChatEditorHint.kt [hunk 1]
```diff
@@ -91,7 +94,7 @@ class InlineChatEditorHint {
             location,
             HintManager.HIDE_BY_TEXT_CHANGE or HintManager.HIDE_BY_SCROLLING,
             0,
-            false,
+            true,
```
Comment (Will-ShaoHua) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/inline/InlineChatEditorHint.kt:97]: what will this config `reviveOnEditorChange` impact

**Review by Will-ShaoHua — APPROVED** [review]

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/inline/InlineChatEditorHint.kt (hunk 1 above)
Comment (zixlin7) [reply plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/inline/InlineChatEditorHint.kt]: reverted that, thanks for catching

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/inline/InlineChatEditorHint.kt [hunk 2]
```diff
@@ -19,23 +19,26 @@ import javax.swing.JPanel
 
 class InlineChatEditorHint {
     private val hint = createHint()
+    private val hintXOffset = 20
```
Comment (rli) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/inline/InlineChatEditorHint.kt:22]: can you double check what happens when the UI is scaled really big?

**Review by rli — APPROVED** [review]

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/inline/InlineChatEditorHint.kt (hunk 2 above)
Comment (zixlin7) [reply plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/inline/InlineChatEditorHint.kt:22]: checked on both small window size and big monitor, it looks fine

**Pull request merged**